Index: hostapd-2018-05-21-62566bc2/src/ap/ieee802_11.c
===================================================================
--- hostapd-2018-05-21-62566bc2.orig/src/ap/ieee802_11.c
+++ hostapd-2018-05-21-62566bc2/src/ap/ieee802_11.c
@@ -2223,6 +2223,10 @@ static u16 check_ext_capab(struct hostap
 			sta->qos_map_enabled = 1;
 	}
 #endif /* CONFIG_INTERWORKING */
+	if (ext_capab_ie_len >= 4) {
+		if (ext_capab_ie[3] & 0x08)
+			sta->bss_transition_supported = 1;
+	}
 
 	if (ext_capab_ie_len > 0) {
 		sta->ecsa_supported = !!(ext_capab_ie[0] & BIT(2));
Index: hostapd-2018-05-21-62566bc2/src/ap/sta_info.h
===================================================================
--- hostapd-2018-05-21-62566bc2.orig/src/ap/sta_info.h
+++ hostapd-2018-05-21-62566bc2/src/ap/sta_info.h
@@ -117,6 +117,7 @@ struct sta_info {
 	unsigned int power_capab:1;
 	unsigned int agreed_to_steer:1;
 	unsigned int hs20_t_c_filtering:1;
+	unsigned int bss_transition_supported:1;
 
 	u16 auth_alg;
 
